nodes/vector/vector_polar_out.py

```diff
@@ -54,9 +54,9 @@ class VectorPolarOutNode(bpy.types.Node, SverchCustomTreeNode):
     ]
 
     def coordinate_changed(self, context):
-        self.outputs["z"].hide = self.coordinates == "theta"
-        self.outputs["theta"].hide = self.coordinates == "z"
-        updateNode(self, context)
+        if self.coordinates != self.outputs[2].name:
+            self.outputs.remove(self.outputs[2])
+            self.outputs.new('StringsSocket', self.coordinates)
```

Comment:
I agree with @zeffii , that it is better to rename the socket than to remove and create new one, because this way existing connections will remain. Probably this is more expected behaivour from the user's point of view.

Reply:
@portnov  .. i think replace_socket keeps track of the old onnection and reconnects..

this goes into: 

    def replace_socket(self, new_type, new_name=None):
        """Replace a socket with a socket of new_type and keep links,
        return the new socket, the old reference might be invalid"""
        return replace_socket(self, new_type, new_name)

this:

https://github.com/nortikin/sverchok/blob/master/data_structure.py#L611

Reply:
So much comments for so small change. ^_^
According all that was suggested, I also have an idea. What about such code?

```Python
    def coordinate_changed(self, context):
        self.outputs[2].name = self.coordinates
```
It looks short and clear, isn't it?
And connections are keeping.

Reply:
please use `.replace_socket(....)` as documented in *data_structure.py*, it handles also the freezing of the node tree. That code is there for a reason :)

Reply:
@portnov @zeffii Is it okay? As I see there is no necessary in `if` function at all.
```Python
    def coordinate_changed(self, context):
        replace_socket(self.outputs[2], 'StringsSocket', new_name = self.coordinates)
```

Reply:
no. use `self.outputs[2].replace_socket("self.coordinates")`  
- if socket type stays the same, it doesn't need to be passed
- if only thing that changes is the name, that's all you pass :)

Reply:
correction? this should work :)  
```python
self.outputs[2].replace_socket("StringsSocket", self.coordinates)
```